### DISCUSSION TITLE
Multiple belong_to's to same entity

### DISCUSSION THREAD

**Discussion by somehowchris** [diesel-rs/diesel#2699] [Ideas]
Hey :),

I'm somwhat new to diesel and rust but I couldn't find the feature related to the problem but I shouldn't be the only one using some setup like the following.

For a school project we have a currency, a currency pair consisting out of two currencies and market data belonging to a currency pair. The main problem is the `#blongs_to` part which I populated with the entity and the foreign_key as needed twice in the currency pair entity but diesel returns an errors at compile time because these two seem to conflict.

```rust
use diesel::prelude::*;
use crate::database::schema::currency_pair;
use crate::currency::entity::CurrencyEntity;

#[derive(Identifiable, Queryable, Associations, serde::Serialize, serde::Deserialize)]
#[serde(rename_all(serialize = "camelCase", deserialize = "snake_case"))]
#[table_name = "currency_pair"]
#[belongs_to(CurrencyEntity, foreign_key="base_currency_id")]
#[belongs_to(CurrencyEntity, foreign_key="secondary_currency_id")]
pub struct CurrencyPairEntity {
    id: i32,
    created_at: chrono::NaiveDateTime,
    updated_at: chrono::NaiveDateTime,
    trading_enabled: bool,
    base_currency_id: i32,
    secondary_currency_id: i32,
    name: String,
}
```

If I'm totally off just RTFM me. But as I couldn't find a solution heres a proposal:
Give every belongs_to part a `name` attribute which can be set manually just like the `foreign_key`. That again cloud be used at load calls where you would do `belonging_to(posts, "<previous given name>")`

**Comment by weiznich** [comment]
This is currently not supported for the reasons listed in https://github.com/diesel-rs/diesel/issues/2142. Also we do not plan to add support for this anytime soon as for the ambiguity reasons listed in the linked issue + the alternative is clearer and not really longer than having the foreign key explicitly on each call to `belonging_to`.
